pkg/operator/reports.go

```diff
@@ -344,7 +344,15 @@ func getReportPeriod(now time.Time, logger log.FieldLogger, report *metering.Rep
 // according the report's schedule. If the next scheduled reporting period
 // hasn't elapsed, runReport will requeue the resource for a time when
 // the period has elapsed.
+// Report expiration is checked before running, report is deleted if expired.
 func (op *Reporting) runReport(logger log.FieldLogger, report *metering.Report) error {
+	// check if the report is expired
+	if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
+		deleteReport(report, op)
+		return nil
+	} else {
+		op.logger.Infof("not deleting report: %s, it's not expired", report.Name)
+	}
```

Comment:
You can deflate this to only using an `if` statement as the `else` clause is redundant.

Reply:
yes, was for debug